File: chap3/MesurePrincipale.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>
#include <cmath>

using namespace std;

int main()
{
   double alpha,principale,n;
   cin >> alpha;
   n = floor(alpha / (2 * M_PI));
   principale = alpha - 2 * n * M_PI;
   if (principale > M_PI)
   {
      principale = principale - 2 * M_PI;
   }
   cout << principale << endl;
}

File: chap3/MesurePrincipale.java
// Ces programmes sont sous licence CeCILL-B V1.
class MesurePrincipale {

  public static void main (String [] args) {
    double alpha,principale,n;
    alpha = Isn.readDouble();
    n = Math.floor(alpha / (2 * Math.PI));
    principale = alpha - 2 * n * Math.PI;
    if (principale > Math.PI) {
      principale = principale - 2 * Math.PI;}
    System.out.println(principale);}}
